strawberryfields/gbs/plot.py

```diff
@@ -0,0 +1,235 @@
+        e_y.append(None)
+
+    return {"x": e_x, "y": e_y}
+
+
+web_green1 = '#84ba5b'
+web_green_light1 = '#e0edd6'
+web_green2 = '#3e9651'
+web_red1 = '#d35e60'
+web_red2 = '#cc2529'
+sf_cyan = '#3e9bb7'
+sf_cyan_light = '#6eb4c9'
+sf_lightgreen = '#daf0e9'
+sf_red = '#d35e60'
+sf_grey = '#efefef'
+light_grey = '#CDCDCD'
+
+graph_node_colour = web_green2
+graph_edge_colour = light_grey
+subgraph_node_colour = web_red2
+subgraph_edge_colour = web_red2
+
+graph_node_size = 14
+subgraph_node_size = 16
+
+
+def plot(graph: nx.Graph, subgraph: Optional[list] = None, size: float = 500) -> None:
+    """ Creates a plot.ly plot of the input graph.
+
+    The graph layout is fixed to be the Kamada-Kawai layout with an aspect ratio of 1:1. The
+    function can plot just the input graph or the graph with a specified subgraph highlighted.
+    The function uses the standard colour theme of green nodes, grey edges, and red highlighted
+    subgraph.
+
+    **Example usage**:
+        >>> graph = nx.complete_graph(10)
+        >>> fig = plot(graph, [0, 1, 2, 3])
+        >>> fig.show()
+
+        Args:
+            graph (nx.Graph): input graph
+            subgraph (list): list of nodes comprising the subgraph to highlight
+            size (dict): size of the plot
+
+        Returns:
+             plot.ly graph object
+        """
+
+    s = graph.subgraph(subgraph)
+    l = nx.kamada_kawai_layout(graph)
+
+    g_nodes = go.Scatter(
+        **_node_coords(graph, l),
+        mode='markers',
+        hoverinfo='text',
+        marker=dict(color=graph_node_colour, size=graph_node_size, line_width=2)
+    )
+
+    g_edges = go.Scatter(
+        **edge_coords(graph, l),
+        line=dict(width=1, color=graph_edge_colour),
+        hoverinfo='none',
+        mode='lines'
+    )
+
+    g_nodes.text = [str(i) for i in graph.nodes()]
+
+    layout = go.Layout(showlegend=False, hovermode='closest',
+                       xaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
+                       yaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
+                       margin=dict(b=0, l=0, r=0, t=25),
+                       height=size,
+                       width=size,
+                       plot_bgcolor='#ffffff'
+                       )
+
+    if subgraph:
+
+        s_edges = go.Scatter(
+            **edge_coords(s, l),
+            line=dict(width=2, color=subgraph_edge_colour),
+            hoverinfo='none',
+            mode='lines'
+        )
+
+        s_nodes = go.Scatter(
+            **_node_coords(s, l),
+            mode='markers',
+            hoverinfo='text',
+            marker=dict(color=subgraph_node_colour, size=subgraph_node_size, line_width=2)
+        )
+
+        s_nodes.text = [str(i) for i in s.nodes()]
+
+        f = go.Figure(data=[g_edges, s_edges, g_nodes, s_nodes], layout=layout)
+
+    else:
+        f = go.Figure(data=[g_edges, g_nodes], layout=layout)
+
+    return f
```

Comment:
Code looks good! Although I think it would be good to be able to input a go.Figure for the user if they want to add to a current graph

Reply:
Is that a common use-case?

Reply:
For now, I would rather avoid having a figure as input, since to me this is a very unnatural thing to do. For example, it's not clear to me that we can't also add to the graph in some other method. Also, I agree with Josh, not sure how common this is

Reply:
Note that this is a very common use case with matplotlib, but I'm not sure if it carries across to plot.ly

Reply:
So my understanding was that the use case in matplotlib is:

- User already has some axes, potentially with stuff already plotted on
- User calls e.g. `nx.draw` and has the axes as an argument.
- The result is that their axes have been added onto with the specified graph

Is that what you are thinking @josh146? And then I guess we should decide if that makes sense and is something we can do in Plotly.

Reply:
In person discussion with @ixfoduap: decided _not_ to have plotly figures as an input argument, and let's wait and see based on future feedback. @josh146 let us know if you agree with that!

Reply:
I agree. While passing axes/figure is a very common matplotlib pattern, I'm not sure the same is true with plot.ly. We can re-evaluate if we have user feedback down the line asking for it.